anvil/src/drawing.rs

```diff
@@ -103,34 +105,42 @@ where
         0
     }
 
-    fn geometry(&self) -> Rectangle<i32, Logical> {
-        Rectangle::from_loc_and_size(self.position, self.size)
+    fn geometry(&self, scale: impl Into<Scale<f64>>) -> Rectangle<i32, Physical> {
+        let scale = scale.into();
+        Rectangle::from_loc_and_size(
+            self.position.to_f64().to_physical(scale).to_i32_round(),
+            self.size.to_f64().to_physical(scale).to_i32_round(),
+        )
```

Comment:
Yeah, that was the reason why I changed it everywhere. So that each element could choose how it rounds. But of course that is a bit limited given that draw_surface_tree is opinionated on how it rounds. But elements doing custom rendering could still benefit from it.